provider/ansible/gcp_utils.py

```diff
@@ -78,52 +78,107 @@ def __init__(self, module, product):
         self._validate()
 
     def get(self, url, body=None, **kwargs):
-        kwargs.update({'json': body, 'headers': self._headers()})
-        try:
-            return self.session().get(url, **kwargs)
-        except getattr(requests.exceptions, 'RequestException') as inst:
-            self.module.fail_json(msg=inst.message)
+        """
+        This method should be avoided in favor of full_get
+        """
+        kwargs.update({'json': body})
+        return self.full_get(url, **kwargs)
 
     def post(self, url, body=None, headers=None, **kwargs):
-        if headers:
-            headers = self._merge_dictionaries(headers, self._headers())
-        else:
-            headers = self._headers()
+        """
+        This method should be avoided in favor of full_post
+        """
+        kwargs.update({'json': body, 'headers': headers})
+        return self.full_post(url, **kwargs)
+
+    def post_contents(self, url, file_contents=None, headers=None, **kwargs):
+        """
+        This method should be avoided in favor of full_post
+        """
+        kwargs.update({'data': file_contents, 'headers': headers})
+        return self.full_post(url, **kwargs)
 
+    def delete(self, url, body=None):
+        """
+        This method should be avoided in favor of full_delete
+        """
+        kwargs = {'json': body}
+        return self.full_delete(url, **kwargs)
+
+    def put(self, url, body=None):
+        """
+        This method should be avoided in favor of full_put
+        """
+        kwargs = {'json': body}
+        return self.full_put(url, **kwargs)
+
+    def patch(self, url, body=None, **kwargs):
+        """
+        This method should be avoided in favor of full_patch
+        """
+        kwargs.update({'json': body})
+        return self.full_patch(url, **kwargs)
+    
+    def full_get(self, url, params=None, **kwargs):
+        """
+        This method fully mimicks the requests API
+        """
+        kwargs['headers'] = self._set_headers(kwargs.get('headers'))
         try:
-            return self.session().post(url, json=body, headers=headers)
+            return self.session().get(url, params=params, **kwargs)
         except getattr(requests.exceptions, 'RequestException') as inst:
-            self.module.fail_json(msg=inst.message)
+            self.module.fail_json(msg=str(inst))
```

Comment:
It looks like `requests.exceptions.RequestException` may contain both the request and the response:

https://2.python-requests.org/en/master/_modules/requests/exceptions/#RequestException

Many requests will contain sensitive data like an auth token -- is this one of those cases where we need to be careful about over logging?

Reply:
That's definitely why I had it that way in the first place! I'll change it back + put a comment to explain why (so I don't go through this whole thing again later on)